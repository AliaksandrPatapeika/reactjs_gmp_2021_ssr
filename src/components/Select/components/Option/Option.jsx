/* eslint-disable jsx-a11y/label-has-for */
import PropTypes from 'prop-types';
import React from 'react';
import {components} from 'react-select';

const Option = (props) => {
  const {isSelected, label} = props;

  return (
    <div>
      {/* // PATTERN: Spreading props on DOM elements */}
      <components.Option {...props}>
        <input
          type="checkbox"
          checked={isSelected}
          onChange={() => null}
        />
        {' '}
        <label>{label}</label>
      </components.Option>
    </div>
  );
};

Option.propTypes = {
  label: PropTypes.string.isRequired,
  isSelected: PropTypes.bool.isRequired
};

export default Option;
