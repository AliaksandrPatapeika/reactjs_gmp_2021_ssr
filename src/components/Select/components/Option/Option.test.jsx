import {noop} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import Option from './Option';

describe('Option', () => {
  test('renders correctly', () => {
    const option = renderer
      .create(<Option isSelected label="label" getStyles={noop} cx={noop} />)
      .toJSON();

    expect(option).toMatchSnapshot();
  });
});
