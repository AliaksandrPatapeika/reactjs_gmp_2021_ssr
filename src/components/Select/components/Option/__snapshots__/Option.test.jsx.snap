// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders correctly 1`] = `
<div>
  <div
    className="css-o8533i-Option"
  >
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
     
    <label>
      label
    </label>
  </div>
</div>
`;
